src/GameLogic/Entities/Iceblock.ts

```diff
@@ -54,18 +58,25 @@ export class Iceblock implements Entity {
 		}
 
 		if (this.melting) {
-			this.containedEntity.x = this.x;
-			this.containedEntity.y = this.y;
-			this.containedEntity.prevX = this.x;
-			this.containedEntity.prevY = this.y;
-
-			level.entities.addEntity(this.containedEntity);
+			if (this.containedEntity !== undefined) {
+				this.containedEntity.x = this.x;
+				this.containedEntity.y = this.y;
+				this.containedEntity.prevX = this.x;
+				this.containedEntity.prevY = this.y;
+
+				level.entities.addEntity(this.containedEntity);
```

Comment:
Please also set `this.containedEntity` to undefined after this line. This will avoid problems in the future, if we add pooling for other entities or have some other dispose mechanism.